script.js

```diff
@@ -1,4 +1,125 @@
+//user name to personalize the game
+var userName = "";
+
+//number of times player wins vs computer wins vs draws
+var mode = "";
+
+//set number of player wins, computer wins, draws to 0
+var numPlayerWins = 0;
+var numComputerWins = 0;
+var numDraws = 0;
+
+//game mode normal
+var mode = "normal";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //computer choice
+  var comChoice = generateComChoice();
+  console.log(comChoice);
+  console.log(input);
+
+  console.log(userName);
+  if (userName == "") {
+    userName = input;
+    return `${userName}, let's play!`;
+  }
+  console.log(userName);
+
+  if (input == `reversePlay`) {
+    mode = `reverse`;
+    return "reverse mode activated";
+  } else if (input == `normalPlay`) {
+    mode = `normal`;
+    return "back to normal mode";
+  }
```

Comment:
I really like the code here in your main function because of the following things that you did which I think are great qualities to keep up

1. It is great that you are able to segment out your logic into helper functions that you manage to integrate inside your main function. By doing so, you have made your code much readable, which is often the next level that you would want to achieve once you are able to write code that works

2. I really like your use of console.logs, which I assume was used by you to debug your code. 

3. It is great to see that you are able to control the mode of your game through the global mode variable and direct the user to different game logic